### PR TITLE
Fix parse error on statement-less files with only newlines, add a warning for empty files.

### LINKED ISSUE #52028: [GDScript 2.0] [regression] parse error on statement-less file
### Godot version

master #fb176d5f6e6dc2b43c630a041a8ab7d737173ec4

### System information

manjaro linux

### Issue description

While empty (0 bytes) file will get parsed successfully, the following files will not (they did in Godot `3.x`):

- newline, newlines:

```


```

 - comment + newline

```
# asd

```

Error:
```
SCRIPT ERROR: Parse Error: Unexpected "Newline" in class body.
```

### Steps to reproduce

compile

### Minimal reproduction project

_No response_

### PR DESCRIPTION
**Pull request by ThreeRhinosInAnElephantCostume**
Fixes #52028 and adds a warning on parsing empty gdscript files.

### CODE REVIEW

File: modules/gdscript/gdscript_warning.h [hunk 1]
```diff
@@ -67,7 +67,8 @@ class GDScriptWarning {
 		STANDALONE_TERNARY, // Return value of ternary expression is discarded.
 		ASSERT_ALWAYS_TRUE, // Expression for assert argument is always true.
 		ASSERT_ALWAYS_FALSE, // Expression for assert argument is always false.
-		REDUNDANT_AWAIT, // await is used but expression is synchronous (not a signal nor a coroutine).
+		REDUNDANT_AWAIT, // Await is used but expression is synchronous (not a signal nor a coroutine).
```
Comment (vnen) [inline modules/gdscript/gdscript_warning.h:70]: Not sure why you changed this. `await` is a keyword, so it should not be capitalized.

**Comment by vnen** [comment]
Seems okay apart from my comment. If you can also add a test file for this it would be great.

**Review by vnen — APPROVED** [review]
LGTM

**Comment by mhilbrunner** [comment]
I think the changes are simple enough that they ought to be squashed into one commit? :)

**Comment by ThreeRhinosInAnElephantCostume** [comment]
> I think the changes are simple enough that they ought to be squashed into one commit? :) Besides that it looks good to me too.

Done.

**Review by vnen — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Some of the tests added in this PR seem to fail randomly on Windows in the CI builds, with this output:
```
===============================================================================
.\modules\gdscript\tests\gdscript_test_runner_suite.h(43):
ERROR: Condition "(type & 0xFF) >= Variant::VARIANT_MAX" is true. Returning: ERR_INVALID_DATA
TEST SUITE: [Modules][GDScript]
   at: decode_variant (core\io\marshalls.cpp:106)
TEST CASE:  Script compilation and runtime

modules\gdscript\tests\gdscript_test_runner.cpp(180): ERROR: CHECK( result.passed ) is NOT correct!
  values: CHECK( false )
  logged: "D:/a/godot/godot/modules/gdscript/tests/scripts/parser/warnings/empty_file.notest.gd"
          ">> WARNING
>> Line: 1
>> EMPTY_FILE
>> Empty script file.
"

modules\gdscript\tests\gdscript_test_runner.cpp(180): ERROR: CHECK( result.passed ) is NOT correct!
  values: CHECK( false )
  logged: "D:/a/godot/godot/modules/gdscript/tests/scripts/parser/warnings/empty_file_newline.notest.gd"
          ">> WARNING
>> Line: 1
>> EMPTY_FILE
>> Empty script file.
"

.\modules\gdscript\tests\gdscript_test_runner_suite.h(47): FATAL ERROR: REQUIRE( fail_count == 0 ) is NOT correct!
  values: REQUIRE( 2 == 0 )
  logged: Make sure `*.out` files have expected results.
          All GDScript tests should pass.

===============================================================================
```

It seems to be non-deterministic as restarting the build often lets it pass.

**Comment by ThreeRhinosInAnElephantCostume** [comment]
> Some of the tests added in this PR seem to fail randomly on Windows in the CI builds, with this output:

Fixed in #52643

**Pull request merged**